src/index.ts

```diff
@@ -25,16 +25,21 @@ swdcTracker.initialize = async (swdcApiHost: string, namespace: string, appId: s
     const result = await get("/plugins/config");
     const tracker_api_host = result.data.tracker_api;
     const tracker_url_scheme = result.data.tracker_url_scheme || "https";
-    // initialize snowplow tracker
-    // endpoint, protocol, port, method, buffer, retry number (max of 2)
-    const completionHandler = function(error?: any, response?: any) {
-      if (error) {
-        console.error("swdc-tracker unstruct track event error", error);
-      }
-    }
-    const e = gotEmitter(tracker_api_host, tracker_url_scheme, 443, HttpMethod.POST, 0, 2, undefined, completionHandler);
 
-    swdcTracker.spTracker = tracker([e], namespace, appId, false)
+    swdcTracker.spTracker = newTracker({
+      namespace: namespace,
+      appId: appId,
+      encodeBase64: false
+    }, {
+      endpoint: tracker_api_host,
+      protocol: tracker_url_scheme,
+      eventMethod: "post",
+      bufferSize: 1,
+      keepalive: true,
+      onRequestFailure(failure, response) {
+        console.error("swdc-tracker failed to send event", JSON.stringify({ failure, response }));
```

Comment:
Need to update this. The failure contains the full event payload. That's probably too much to log.